### ISSUE TITLE
Metrics Requirements Checklist (Project Kick-off)

### ISSUE THREAD

**Issue by adamlofting** [mozilla/id.webmaker.org#56]
- [ ] We **need** to be able to count/measure:
 * Page views for all routes
 * Any important user interactions that don't trigger a new route
 * Validation errors
 * .... add further requirements here

- [ ] We **would like** to be able to count/measure:
 * Front-end performance metrics
 * .... add further requirements here

- [x] We have investigated (or asked Adam) which of these items are already being measured, and which need new 'instrumentation'
 * This is a new repo, there is no existing instrumentation

- [ ] TODO: File instrumentation tickets, once questions above are answered
 * *Link to ticket(s)*

- [ ] We have decided on the level of reporting needed, on a scale of: '_Ask Adam two weeks later how it went_' to '_We need a realtime dashboard and my phone to ring if this number changes by more than half a percentage point_'
 * *Reporting requirements are...*

- [ ] Decide what 'success looks like' by setting goals for the numbers you want to track
 * *We hope to X new Ys...*

- [ ] If you are planning to run A/B tests or similar, please briefly outline the testing strategy. 
 * No A/B tests planned for launch, but capacity for A/B testing will be required
 * Need to include Optimizely tags, and check this is built in a way that works with testing

**Comment by adamlofting** [comment]
cc/ @thisandagain for reporting requirements.

Assuming instrumentation requirement = all the things :zap:

**Comment by thisandagain** [comment]
From a product POV, my main interests are:

- Conversion rates
    - UV to Sign Up
    - UV to Log In
- Abandon rate
    - % of users who start but never complete sign up
    - % of users who start but never complete log in
- Abandon funnel
    - % of users who exit the funnel at each stage of sign up (by view is fine, but by input is better)
    - % of users who exit the funnel at each stage of sign up (by view is fine, but by input is better)
- Error frequency
    - Break down of validation errors by message and both % and # of occurrences
    - Break down of HTTP errors (404s, 500s, etc) created by both the API and WWW interface

**Comment by adamlofting** [comment]
Should we be treating a users time spent on these pages as part of their webmaker product experience? Given some users will come in from teach, and bounce back out to teach? Or should we treat id.w.o as it's own service (like persona) with it's own internal stats separate from those of webmaker?

There are some complications here in terms of teach.m.o user behaviors artificially distorting (up or down) the stats for Webmaker, unless we split these out somehow.

**Comment by adamlofting** [comment]
Following discussion with @thisandagain, 

The plan is to measure id.webmaker.org as a standalone product for it's 'internal metrics'.
*e.g. Conversion rate once people land on the id.webmaker.org signup*

Separate products and tools that use this ID system will measure their own conversion rate without needing to know the internal metrics of id.webmaker.org. 
*e.g. 1000 people visit teach.mozilla.org, X% click to create an account and are routed out to id.w.o, Y% return with account creation successful.*

This allows optimization of id.w.o to benefit all sites that make user of this authentication system. And sites to manage their own conversion rates without overlapping data about unrelated users (a teacher who isn't planning to use the tool or vice versa).

I'll work through the implications for this and start writing specs and filing tickets.

**Comment by adamlofting** [comment]
Also notes to self todo tomorrow: 
- [ ] What happens to the referral API we build last year? Do we port it across?
- [ ] How do we know in the database, which users have engaged with which sites. So that we can count the number of webmaker users as distinct from the number of webmaker accounts (which will be a bigger number)

**Comment by adamlofting** [comment]
See actions, and remaining questions here: https://github.com/mozilla/id.webmaker.org/issues/77
